src/rand.rs

```diff
@@ -0,0 +1,35 @@
+#![cfg(feature = "rand")]
+
+use crate::Decimal;
+use rand::{
+    distributions::{Distribution, Standard},
+    Rng,
+};
+
+impl Distribution<Decimal> for Standard {
+    fn sample<R>(&self, rng: &mut R) -> Decimal
+    where
+        R: Rng + ?Sized,
+    {
+        Decimal::from_parts(
+            rng.next_u32(),
+            rng.next_u32(),
+            rng.next_u32(),
+            rng.gen(),
+            rng.next_u32(),
+        )
+    }
+}
+
+#[test]
+fn has_random_decimal_instances() {
+    let mut rng = rand::rngs::OsRng;
+    let random: Decimal = rng.gen();
+    for _ in 0..100 {
+        let local_random: Decimal = rng.gen();
+        if random != local_random {
```

Comment:
This is a bit of an odd test to me! Do we expect the first, say 50 iterations to generate the same number as `random`?
Another idea: would it be possible to generate 100 numbers and check they're all different?

Reply:
Nope, the test is more like "Is the implementation generating random number at all?".

> Another idea: would it be possible to generate 100 numbers and check they're all different?

This will make CI intermittent because sometimes (even rare) it is possible to have at least one repeated element.

Reply:
Let me know if the updated test improved the situation.

Reply:
Fair enough - let's give it a go with the updated test and if it ends up being problematic we can adjust it. Thank you!